nomenclate/core/configurator.py

```diff
@@ -38,150 +25,214 @@ def __init__(self, path=None):
-        """ Function to check if the config section/subsection data exists
-        """
-        data_query = None
-        if not subsection:
-            data_query = self.data.get(section, '')
-        else:
-            data_query = self.data.get(section, {}).get(subsection, '')
-        if not data_query:
-            raise IOError('Section: %s and Subsection: %s not found in current config file...' % (section, subsection))
-        return True
+    def get_path_entry_from_config(self, query_path):
+        if not query_path:
+            return list(self.data)
+        cur_data = self.data
+        for child in query_path:
+            cur_data = cur_data.get(child)
+        return cur_data
 
-    def get_section(self, section=None, raw=False):
-        """
-        Returns (dict): dictionary of the entire contents of a specific section
-        """
-        return self.get(section, raw=raw)
+    @classmethod
+    def validate_config_file(cls, config_file):
+        if not os.path.isfile(config_file):
+            raise IOError('File %s is not a valid yml, ini or cfg file or does not exist' % config_file)
 
-    def get_subsection(self, section=None, subsection=None, raw=False):
-        """ Return the given section/subsection pair as a list
-        Args:
-            section (str): section to query
-            subsection (str): subsection to query
-        Returns (dict): list of the subsection results
-        """
-        return self.get(section=section, subsection=subsection, raw=False)
+        elif os.stat(config_file).st_size > 0:
+            raise IOError('File %s is empty' % config_file)
 
-    def get_subsection_as_dict(self, section=None, subsection=None):
-        """ Return the given section/subsection pair as a dictionary
-        Args:
-            section (str): section to query
-            subsection (str): subsection to query
-        Returns (dict): dictionary of the subsection results
-        """
-        return {section: {subsection: self.get_subsection(section=section, subsection=subsection)}}
+        with open(config_file, 'r') as f:
+            if yaml.load(f) is None:
+                raise IOError('No YAML config was found in file %s' % config_file)
 
-    def get_subsection_as_str(self, section=None, subsection=None):
-        """ Return the given section/subsection pair as a string
+    def validate_query_path(self, query_path):
+        """ Determines whether the query path given is found in the current dataset
         Args:
-            section (str): section to query
-            subsection (str): subsection to query
-        Returns (str): string of the subsection results
+            query_path [str]: list of query paths to traverse
+        Returns (bool): True or raise IndexError on non found query
         """
-        return self.get(section=section, subsection=subsection, raw=True)
+        cur_data = self.data
+        for path in query_path:
+            cur_data = cur_data.get(path)
```

Comment:
I'm doing that on purpose, let's call it ghetto recursion, it follows the query_path down through the dictionary and keeps replacing the dictionary with the sub dictionary.  I could just replace this with get functionality though or something like that...